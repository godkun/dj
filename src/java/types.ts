import { JavaClazz } from './core/clazz'

export interface IType {
  tsType: string
  javaType: string
  generic?: Array<JavaClazz | IType>
}

export interface IJavaClazzField {
  name: string
  type: IType
  comment?: string
}

export type TServiceMethod = {
  [k in string]: IMethodMeta
}

export type IMethodArgs = {
  name: string
  type: string
}

export interface IMethodMeta {
  args: Array<IMethodArgs>
  ret: string
}
