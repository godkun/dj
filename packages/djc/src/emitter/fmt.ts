import p from 'prettier'

/**
 * format generate code
 * @param str
 */
export default function fmt(str: string) {
  return p.format(str, {
    trailingComma: 'es5',
    tabWidth: 2,
    semi: false,
    singleQuote: true,
    parser: 'typescript',
  })
}
