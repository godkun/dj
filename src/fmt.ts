import p from 'prettier'

export default function fmt(str: string) {
  return p.format(str, {
    trailingComma: 'es5',
    tabWidth: 2,
    semi: false,
    singleQuote: true,
    parser: 'typescript',
  })
}
