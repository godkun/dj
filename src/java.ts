import Deps from './java-deps'

export default class Base {
  deps: Deps = new Deps()

  cls2Path(cls: string) {
    return cls.replace(/\./g, '/')
  }

  buildComment(comment?: string) {
    return comment
      ? `
    /**
    * ${comment}
    */
   `
      : ''
  }

  get mott() {
    return `
      /**
       * auto generated by dubbo dj
       * ~~~ mechine coding ðŸ’— ~~~~
       */
    `
  }
}
