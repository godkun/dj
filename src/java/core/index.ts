/**
 * 通过构造Java Style code的DSL来自动生成 ts，java， go的dubbo框架的调用代码
 */

import Clazz from './clazz'
import Service from './service'
import {
  Integer,
  String,
  Boolean,
  List,
  Set,
  Collection,
  Map,
  boolean,
  int,
  short,
  Short,
  byte,
  Byte,
  long,
  Long,
  double,
  Double,
  float,
  Float,
  char,
  Char,
  chars,
  Character,
  Enumeration,
  Iterator,
  Currency,
  Dictionary,
  HashMap,
} from './java-type'

import Enum from './enum'

export default {
  Enum,
  Clazz,
  Service,

  Boolean,
  boolean,

  Integer,
  int,

  short,
  Short,

  byte,
  Byte,

  long,
  Long,

  double,
  Double,

  float,
  Float,

  String,
  char,
  Char,
  chars,
  Character,

  List,
  Set,
  Collection,
  Iterator,
  Enumeration,
  Map,
  HashMap,
  Dictionary,
  Currency,
}
