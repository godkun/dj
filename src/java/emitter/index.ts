import Clazz from './clazz'
import Enum from './enum'
import Service from './service'
import AbstractService from './service-abstract'

export { Enum, Clazz, Service, AbstractService }
