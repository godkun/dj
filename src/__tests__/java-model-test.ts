import fmt from '../fmt'
import java from '../'

const user = java
  .model('org.apache.dubbo.js.User')
  .field('id', java.Integer, '用户id')
  .field('name', java.String, '用户名')
  .field('isMale', java.Boolean)

const h = java
  .model('org.apache.dubbo.js.service.User')
  .field('id', java.Integer)

const product = java
  .model('org.apache.dubbo.js.Product')
  .field('id', java.Integer, '商品')
  .field('price', java.Integer, '价格')
  .field('color', java.String, '颜色')
  .field('user', user, '管理用户')
  .field('user2', user, 'xx')
  .field('h', h)

it('test user inf and class', () => {
  expect(user).toMatchSnapshot()
  expect(JSON.stringify(user, null, 2)).toMatchSnapshot()
  expect(fmt(user.inf)).toMatchSnapshot()
  expect(fmt(user.bean)).toMatchSnapshot()
})

it('test product import and rename', () => {
  expect(fmt(product.inf)).toMatchSnapshot()
  expect(fmt(product.bean)).toMatchSnapshot()
})
