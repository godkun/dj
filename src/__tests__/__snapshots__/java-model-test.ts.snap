// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test product import and rename 1`] = `
"/**
 * auto generated by dubbo dj
 * ~~~ mechine coding 💗 ~~~~
 */

import User from 'org/apache/dubbo/js/User'

import User1 from 'org/apache/dubbo/js/service/User'

export interface IProduct {
  /**
   * 商品
   */
  id: number

  /**
   * 价格
   */
  price: number

  /**
   * 颜色
   */
  color: string

  /**
   * 管理用户
   */
  user: User

  /**
   * xx
   */
  user2: User
  h: User1
}
"
`;

exports[`test product import and rename 2`] = `
"/**
 * auto generated by dubbo dj
 * ~~~ mechine coding 💗 ~~~~
 */

import User from 'org/apache/dubbo/js/User'

import User1 from 'org/apache/dubbo/js/service/User'

export default class Product {
  /**
   * 商品
   */
  id: number

  /**
   * 价格
   */
  price: number

  /**
   * 颜色
   */
  color: string

  /**
   * 管理用户
   */
  user: User

  /**
   * xx
   */
  user2: User
  h: User1

  constructor(props: IProduct) {
    this.id = props.id
    this.price = props.price
    this.color = props.color
    this.user = props.user
    this.user2 = props.user2
    this.h = props.h
  }

  __fields2Java() {
    java('org.apache.dubbo.js.Product', {
      id: java.Integer(this.id),
      price: java.Integer(this.price),
      color: java.String(this.color),
      user: this.user.__fields2Java(),
      user2: this.user2.__fields2Java(),
      h: this.h.__fields2Java(),
    })
  }
}
"
`;

exports[`test user inf and class 1`] = `
Model {
  "cls": "org.apache.dubbo.js.User",
  "clsName": "User",
  "deps": Deps {
    "map": Map {},
  },
  "fields": Array [
    Object {
      "comment": "
    /**
    * 用户id
    */
   ",
      "name": "id",
      "type": Object {
        "javaType": "java.Integer(this.id)",
        "tsType": "number",
      },
    },
    Object {
      "comment": "
    /**
    * 用户名
    */
   ",
      "name": "name",
      "type": Object {
        "javaType": "java.String(this.name)",
        "tsType": "string",
      },
    },
    Object {
      "comment": "",
      "name": "isMale",
      "type": Object {
        "javaType": "java.Boolean(this.isMale)",
        "tsType": "boolean",
      },
    },
  ],
}
`;

exports[`test user inf and class 2`] = `
"{
  \\"deps\\": {
    \\"map\\": {}
  },
  \\"cls\\": \\"org.apache.dubbo.js.User\\",
  \\"fields\\": [
    {
      \\"name\\": \\"id\\",
      \\"type\\": {
        \\"tsType\\": \\"number\\",
        \\"javaType\\": \\"java.Integer(this.id)\\"
      },
      \\"comment\\": \\"\\\\n    /**\\\\n    * 用户id\\\\n    */\\\\n   \\"
    },
    {
      \\"name\\": \\"name\\",
      \\"type\\": {
        \\"tsType\\": \\"string\\",
        \\"javaType\\": \\"java.String(this.name)\\"
      },
      \\"comment\\": \\"\\\\n    /**\\\\n    * 用户名\\\\n    */\\\\n   \\"
    },
    {
      \\"name\\": \\"isMale\\",
      \\"type\\": {
        \\"tsType\\": \\"boolean\\",
        \\"javaType\\": \\"java.Boolean(this.isMale)\\"
      },
      \\"comment\\": \\"\\"
    }
  ],
  \\"clsName\\": \\"User\\"
}"
`;

exports[`test user inf and class 3`] = `
"/**
 * auto generated by dubbo dj
 * ~~~ mechine coding 💗 ~~~~
 */

export interface IUser {
  /**
   * 用户id
   */
  id: number

  /**
   * 用户名
   */
  name: string
  isMale: boolean
}
"
`;

exports[`test user inf and class 4`] = `
"/**
 * auto generated by dubbo dj
 * ~~~ mechine coding 💗 ~~~~
 */

export default class User {
  /**
   * 用户id
   */
  id: number

  /**
   * 用户名
   */
  name: string
  isMale: boolean

  constructor(props: IUser) {
    this.id = props.id
    this.name = props.name
    this.isMale = props.isMale
  }

  __fields2Java() {
    java('org.apache.dubbo.js.User', {
      id: java.Integer(this.id),
      name: java.String(this.name),
      isMale: java.Boolean(this.isMale),
    })
  }
}
"
`;
